sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -228,11 +204,71 @@ private Tree createFunctionDeclarationTree(@NotNull KtFunction functionElement,
       .collect(Collectors.toList());
   }
 
-  @NotNull
-  private Tree createBinaryExpression(@NotNull KtBinaryExpression element, TreeMetaData metaData) {
+  private Tree createIfTree(TreeMetaData metaData, KtIfExpression element) {
+    Tree condition = createElement(element.getCondition());
+    Tree thenBranch = createElement(element.getThen());
+    Tree elseBranch = createElement(element.getElse());
+    if (condition == null) {
+      throw new ParseException("Invalid 'If' structure. 'condition' cannot be null", metaData.textRange().start());
+    }
+
+    if (thenBranch == null) {
+      // Kotlin allows for a null then branch, which we match to a native since this is not allowed in Slang
+      thenBranch = nullNative(condition.metaData());
+    }
+    return new IfTreeImpl(metaData, condition, thenBranch, elseBranch);
+  }
+
+  private static Tree createIdentifierTree(TreeMetaData metaData, @Nullable String name) {
+    if (name == null) {
+      throw new ParseException("Identifier name cannot be null", metaData.textRange().start());
+    }
+    return new IdentifierTreeImpl(metaData, name);
+  }
+
+  private static Tree nullNative(TreeMetaData metaData) {
+    // We use the metadata of the parent here, as a NULL kind has no range itself
```

Comment:
What's a "null native"? If it's a tree which does not exist, then I don't think we should represent it as an instance of Tree. I would prefer to represent the parent of such a tree as a native. For example, if Kotlin accepts an if statement with no "then" branch, then I think we should represent the whole statement with a NativeTree.

Reply:
Yes, I agree, this is a much better solution. Updated the PR accordingly